### ISSUE TITLE
Discharge issui

### ISSUE THREAD

**Issue by Michaelj15** [TobiasF92/FS22_interactiveControl#6]
Hello,

I don't know if it's an issui or just a wrong textcode in the xml.
What i try to do is to make a discharge button outside of the tractor for a mixerwagon. 
(just like to activate a pto outside the tractor, but this time for unload forage in a cowshed).
I have tried a lot of codes, but i can't activate the function. 
Everything is showing up and i can click the button, but there isn't a function behind it. 
I used the ATTACHERJOINTS_TOGGLE_DISCHARGE and the DISCHARGE_TOGGLE but both without succes.

What i'm doing wrong? 

                    
					<interactiveControl posText="$l10n_actionIC_StopUnloadingImplement" negText="$l10n_actionIC_UnloadImplement" >
                        <clickPoint node="0>0|6|19|7" size="0.05" blinkSpeedScale="0.6" type="OUTDOOR" iconType="TURN_ON" alignToCamera="false" invertZ="true"/>
						<function name="DISCHARGE_TOGGLE"/>
					</interactiveControl>

**Comment by janus579** [comment]
Hi! You have forgot about attacherJoint.
It should look like this:
<function name="ATTACHERJOINTS_TOGGLE_DISCHARGE"> <- this one is for attached machines
	<attacherJoint indicies="1 2 3"/>      <- here you put attacherJoint indexes that you want to use
</function>
:)

**Comment by Michaelj15** [comment]
> Hi! You have forgot about attacherJoint. It should look like this: function name="ATTACHERJOINTS_TOGGLE_DISCHARGE"> <- this one is for attached machines attacherJoint indicies="1 2 3"/> <- here you put attacherJoint indexes that you want to use /function> :)

Thanks for the quick reply! 
For my understanding, what are attacherjoint indices or indexes?
I know that attacherjoints getting numbers from the xml, mostly 1 for back, 2 for front etc, but what are indices? Are these the hydraulic hoses or? Because I typed these already and the button didn't show up. 
![16696414814523277022967006478856](https://user-images.githubusercontent.com/59195589/204287426-fabde38c-086e-460c-9df3-86e8b152774f.jpg)

**Comment by janus579** [comment]
You have to put it in one attacherJoint like I did, unluckily game is crashing when you try to attach something other than a trailer :/

![image](https://user-images.githubusercontent.com/119316289/204288338-86a2eddc-ba92-4656-90c8-f98f5fb9d6fa.png)

**Comment by janus579** [comment]
![image](https://user-images.githubusercontent.com/119316289/204288996-b650a36f-ce6a-4c87-a227-f3e9c6cbbf19.png)
It should look like this, but i recommend to wait with it until next IC update

**Comment by Michaelj15** [comment]
> ![image](https://user-images.githubusercontent.com/119316289/204288996-b650a36f-ce6a-4c87-a227-f3e9c6cbbf19.png) It should look like this, but i recommend to wait with it until next IC update

Thanks!
Yes exactly that's the same problem what I have (black screen) .
No problem, I will wait for the update. Now I know it's not a wrong conflict with another mod or something like that 😁

**Comment by Michaelj15** [comment]
Do i also need to use a indicies for a front implement? Can't get it to work with the code below?

![image](https://user-images.githubusercontent.com/59195589/204813476-05061ea0-6fa7-4ec1-b788-d65c8b6f00df.png)
![image](https://user-images.githubusercontent.com/59195589/204813584-e69afc72-761a-4e16-8ddc-f6438fd42933.png)

**Comment by TobiasF92** [comment]
The Folding function is bugged in release version too, try to use the 1.1-beta:
https://github.com/TobiasF92/FS22_interactiveControl/releases/tag/1.1.0.0-beta1

**Comment by Aim400kgp** [comment]
Has anyone already succeeded in making a button on the outside of the cuagnik to unload bales or grain from a feed wagon or from a trailer or from a baler to unload bales or grain from a trailer?
I'm talking about the Interactive Control mod and the base of the function is DISCHARGE_TOGGLE

**Comment by Aim400kgp** [comment]
> Witam,
> 
> Nie wiem, czy problem, czy po prostu zły kod tekstowy w xml. To, co próbuję zrobić, to zrobić przycisk wyładowczy na zewnątrz ciągnika dla wozu paszowego. (podobnie jak włączenie WOM na zewnątrz ciągnika, ale tym razem do wyładunku paszy w oborze). Uzyskać wielu kodów, ale nie mogę aktywować tej funkcji. Wszystko się wyświetla i mogę użyć przycisku, ale nie ma za tym żadnej funkcji. Użyłem ATTACHERJOINTS_TOGGLE_DISCHARGE i DISCHARGE_TOGGLE, ale oba bez powodzenia.
> 
> Co źle się stało?
> 
> ```
>                 
> 				<interactiveControl posText="$l10n_actionIC_StopUnloadingImplement" negText="$l10n_actionIC_UnloadImplement" >
>                     <clickPoint node="0>0|6|19|7" size="0.05" blinkSpeedScale="0.6" type="OUTDOOR" iconType="TURN_ON" alignToCamera="false" invertZ="true"/>
> 					<function name="DISCHARGE_TOGGLE"/>
> 				</interactiveControl>                   
> ```
Has anyone already succeeded in making a button on the outside of the tractor to unload fodder or bales or grain from a trailer?
I am talking about the mod interactive Control and the name of the function is DISCHARGE_TOGGLE

**Comment by Michaelj15** [comment]
> > Witam,
> > Nie wiem, czy problem, czy po prostu zły kod tekstowy w xml. To, co próbuję zrobić, to zrobić przycisk wyładowczy na zewnątrz ciągnika dla wozu paszowego. (podobnie jak włączenie WOM na zewnątrz ciągnika, ale tym razem do wyładunku paszy w oborze). Uzyskać wielu kodów, ale nie mogę aktywować tej funkcji. Wszystko się wyświetla i mogę użyć przycisku, ale nie ma za tym żadnej funkcji. Użyłem ATTACHERJOINTS_TOGGLE_DISCHARGE i DISCHARGE_TOGGLE, ale oba bez powodzenia.
> > Co źle się stało?
> > ```
> >                 
> > 				<interactiveControl posText="$l10n_actionIC_StopUnloadingImplement" negText="$l10n_actionIC_UnloadImplement" >
> >                     <clickPoint node="0>0|6|19|7" size="0.05" blinkSpeedScale="0.6" type="OUTDOOR" iconType="TURN_ON" alignToCamera="false" invertZ="true"/>
> > 					<function name="DISCHARGE_TOGGLE"/>
> > 				</interactiveControl>                   
> > ```
> 
> Has anyone already succeeded in making a button on the outside of the tractor to unload fodder or bales or grain from a trailer? I am talking about the mod interactive Control and the name of the function is DISCHARGE_TOGGLE

No, unloading didn't work for me. Unfolding is already fixed in the beta. I just wait for an new update, because the game still crashed like the comments above.

**Comment by TobiasF92** [comment]
Can you retest this with current Beta2 Version?
https://github.com/TobiasF92/FS22_interactiveControl/releases/tag/1.1.0.0-beta2

**Comment by janus579** [comment]
> Can you retest this with current Beta2 Version? https://github.com/TobiasF92/FS22_interactiveControl/releases/tag/1.1.0.0-beta2

I have tested it like a week ago, it works properly right now!

**Comment by Michaelj15** [comment]
> Can you retest this with current Beta2 Version? https://github.com/TobiasF92/FS22_interactiveControl/releases/tag/1.1.0.0-beta2

Indeed works wonderful !! Thank you :)

I have two separate questions: 
Sometimes i could activate the buttons when a implement is selected with the G key, but mostly i need to select the tractor for activating the buttons. Do you know what the difference is, or how to solve to activate it when i don't select the tractor?
And is it possible to let the interactive control visible when switching vehicles? When i enter a next vehicle the buttons are always invisible.

**Comment by TobiasF92** [comment]
> Sometimes i could activate the buttons when a implement is selected with the G key, but mostly i need to select the tractor for activating the buttons.
Can you provide me an example? Like a vehicle combination where this happens?

> let the interactive control visible when switching vehicles
Technically no problem, but as i remember correctly it was a wish to deactivate it, while switching
I will collect it for some options in settings, so you can decide what happens :)

**Comment by Michaelj15** [comment]
No problem, gonna try this in the evening. I think it's also the problem in combination with autodrive.
________________________________

**Comment by Michaelj15** [comment]
> > Sometimes i could activate the buttons when a implement is selected with the G key, but mostly i need to select the tractor for activating the buttons.
> 
> Can you provide me an example? Like a vehicle combination where this happens?

I have tested it with a few vehicles:

It happens with a fendt 939, with frontweight and a trailer: See attached picture

![image](https://user-images.githubusercontent.com/59195589/215030091-c7d2a5d2-da90-40a6-b5a4-4b83c97f3065.png)

But also with a planter combination and a fendt scr
![image](https://user-images.githubusercontent.com/59195589/215032309-c32f39ac-477c-475e-8ee6-cb292ff2a925.png)

Also with the new holland below. Strange thing is that with this vehicle also the total combination flips when activating interactive control for the first time. I think it's vehicle related because this is only happening with this one.

![image](https://user-images.githubusercontent.com/59195589/215031227-9ac5168d-27f5-46ae-ade1-022de6218a1c.png)

As far i can see it's only with these three tractors. The other tractors working fine, but they have exact the same xml index :S

**Comment by Kalderone** [comment]
Hello, I wanted to add a mod to the combine, everything turned out great. 
But I have a question: is it possible to connect this script to the animation of the Screw conveyor and when it is turned on for harvesting?
